counsel.el

```diff
@@ -1678,6 +1679,57 @@ When INITIAL-INPUT is non-nil, use it in the minibuffer during completion."
           (format "http://debbugs.gnu.org/cgi/bugreport.cgi?bug=%s"
                   (substring url 1)))))))
 
+(defvar counsel-url-expansions
+  nil
+  "List of (REGEXP . FORMAT) pairs.
+
+`counsel-url-expand' will expand the word at point according to
+FORMAT for the first matching REGEXP.  FORMAT can be either a
+string or a function.  If it is a string, it will be used as the
+format string for the `format' function, with the word at point
+as the next argument.  If it is a function, it will be called
+with the word at point as the sole argument.
+
+For example, a pair of the form:
+  '(\"^BSERV-[0-9]+$\" . \"https://jira.atlassian.com/browse/%s\")
+will expand to URL `https://jira.atlassian.com/browse/BSERV-100'
+when the word at point is BSERV-100.
+
+If the format element is a function, more powerful
+transformations are possible.  As an example,
+  '(\"^issue\\([0-9]+\\)$\" .
```

Comment:
Nitpick: Why not give regexp examples with beginning and end of string, rather than line, anchors?

Reply:
E.g.
```el
\"\\`issue\\([[:digit:]]+\\)\\'\"
```

Reply:
I guess I'm less familiar with emacs regular expressions. ^/$ are just more common and the distinction here seems a little academic. But I suppose \` and \' are more precise, so I'll adjust them.

Reply:
Hence my starting the comment with 'nitpick'. :) I'm just of the opinion that examples ought to be as precise/idiomatic as possible.

Reply:
fair. As an elisp newbie I appreciate the attention you've given this PR.